packages/node/src/integrations/tracing/prisma.ts

```diff
@@ -1,11 +1,58 @@
 import type { Instrumentation } from '@opentelemetry/instrumentation';
 // When importing CJS modules into an ESM module, we cannot import the named exports directly.
 import * as prismaInstrumentation from '@prisma/instrumentation';
-import { SEMANTIC_ATTRIBUTE_SENTRY_ORIGIN, defineIntegration, spanToJSON } from '@sentry/core';
+import { SEMANTIC_ATTRIBUTE_SENTRY_ORIGIN, consoleSandbox, defineIntegration, spanToJSON } from '@sentry/core';
 import { generateInstrumentOnce } from '../../otel/instrument';
+import type { PrismaV5TracingHelper } from './prisma/vendor/v5-tracing-helper';
+import type { PrismaV6TracingHelper } from './prisma/vendor/v6-tracing-helper';
 
 const INTEGRATION_NAME = 'Prisma';
 
+const EsmInteropPrismaInstrumentation: typeof prismaInstrumentation.PrismaInstrumentation =
+  // @ts-expect-error We need to do the following for interop reasons
+  // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
+  prismaInstrumentation.default?.PrismaInstrumentation || prismaInstrumentation.PrismaInstrumentation;
+
+type CompatibilityLayerTraceHelper = PrismaV5TracingHelper & PrismaV6TracingHelper;
+
+function isPrismaV6TracingHelper(helper: unknown): helper is PrismaV6TracingHelper {
+  return !!helper && typeof helper === 'object' && 'dispatchEngineSpans' in helper;
+}
+
+class SentryPrismaInteropInstrumentation extends EsmInteropPrismaInstrumentation {
+  public constructor() {
+    super();
+  }
+
+  public enable(): void {
+    super.enable();
+
+    const prismaInstrumentationObject = (globalThis as Record<string, unknown>).PRISMA_INSTRUMENTATION;
```

Comment:
m: Can we add a comment here explaining what and why we are doing here? For future reference :D